### DISCUSSION TITLE
Attaching a payload to the quadrotor

### DISCUSSION THREAD

**Discussion by eldewany** [microsoft/AirSim#3662] [Support Q&A] [answered]
Hello, i'm wondering if i can attach a certain payload like a camera on a gimbal on a quadrotor, how can I add that given that I already have the 3d models. also, the second thing is about physics and the interaction of the payload weight with the quadrotor physics, how can I define the payload parameters like inertia and weight so they can correctly affect the quadrotor. thanks in advance.

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @eldewany, here's a tutorial that shows how to import a custom drone mesh into AirSim: https://youtu.be/Bp86WiLUC80. Keep in mind that this is only for the visual side of things. As for the physics, AFAIK all you would be able to do right now is add the mass of the payload to the drone's overall mass through [this line in Multirotorparams](https://github.com/microsoft/AirSim/blob/8d21f60e3ac99c4bafa664560f9ea9037228c406/AirLib/include/vehicles/multirotor/MultiRotorParams.hpp#L227) (and you might be able to modify the drone's inertia through the inertia matrix being generated in the same file)
